lint/importShadow_checker.go

```diff
@@ -0,0 +1,61 @@
+package lint
+
+//! Detects when imported package names shadowed in assignments.
+//
+// @Before:
+// import (
+// 	   "fmt"
+//     "math"
+// )
+// func main() {
+//     fmt.Println(math.Pi)
+//     // shadowing math package
+//     math := 10
+//     fmt.Println(len)
+// }
+//
+// @After:
+// import (
+//     "fmt"
+//     "math"
+// )
+// func main() {
+//     fmt.Println(math.Pi)
+//     // change identificator name
+//     value := 10
+//     fmt.Println(value)
+// }
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/go-critic/go-critic/lint/internal/astwalk"
+)
+
+func init() {
+	addChecker(&importShadowChecker{}, attrSyntaxOnly)
```

Comment:
Since this checker uses `types.Package`, it should probably not have `attrSyntaxOnly` attribute.
Generally, it's safer to omit this attribute, at least until we implement #231.